src/lfortran/semantics/ast_common_visitor.h

```diff
@@ -4257,11 +4257,19 @@ class CommonVisitor : public AST::BaseVisitor<Derived> {
             if( !raise_error ) {
                 return false;
             }
-            throw SemanticError("Incorrect number of arguments "
-                                "passed to the " + intrinsic_name + " intrinsic. "
-                                "It accepts at least " + std::to_string(min_args) +
-                                " and at most " + std::to_string(max_args) + " arguments.",
-                                x.base.base.loc);
+            if (min_args == max_args) {
+                throw SemanticError("Incorrect number of arguments "
+                                    "passed to the " + intrinsic_name + " intrinsic. "
```

Comment:
```suggestion
                                    "passed to the '" + intrinsic_name + "' intrinsic. "
```

Since we are updating this, shall we print the name in quotes like `gfortran` does?

```console
% gfortran examples/expr2.f90 
examples/expr2.f90:4:8:

    4 | print *, complex(1)
      |        1
Error: Missing actual argument 'y' in call to 'complex' at (1)
```

Reply:
+1

Reply:
Sounds good to me, in this case it's quite clear what needs to be quoted, so I'll do it.